modules/snowflake-account/provider-snowflake.tf

```diff
@@ -0,0 +1,12 @@
+data "aws_ssm_parameter" "snowflake_password" {
+  count           = local.enabled ? 1 : 0
```

Comment:
I don't know anything about Snowflake. All I did was take the non-standard stuff out of the old `providers.tf` and put it in the new `providers-snowflake.tf` so that in the future we can update `providers.tf` by just copying the new version in. If `providers-snowflake.tf` is broken, I would like to fix that in a separate PR.